src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,42 @@
+import { useRef, useState } from "react";
+import ModalBasic from "./ModalBasic/ModalBasic";
+import styles from "./Modal.module.scss";
+import { useOnClickOutside } from "usehooks-ts";
+
+interface Modal {
+  name: string;
+  subtitle?: string;
+  shareButtons?: boolean;
+}
+
+function Modal({ name, ...rest }: Modal) {
```

Comment:
다른 컴포넌트에서는 함수 선언식에서 바로 export를 해줘서 일관되게 컨벤션을 가져가주면 좋을 것 같네요~
```suggestion
export default function Modal({ name, ...rest }: Modal) {
```